app/.server/parse-markdown.js

```diff
@@ -1,34 +1,37 @@
 import fs from 'node:fs/promises';
 import matter from 'gray-matter';
 import { Marked } from 'marked';
-import markedShiki from 'marked-shiki';
-import { getHighlighter } from 'shiki';
+import { createHighlighter } from 'shiki';
 import etag from './etag';
 import { getPrettyDate } from './get-pretty-date';
 import { replaceVideoTags } from './replace-video-tags';
 
-const highlighter = getHighlighter({
-  langs: ['md', 'sh', 'bash', 'rust', 'plaintext', 'yaml'],
-  themes: ['github-dark-dimmed'],
-});
+let highlighter;
 
 async function parseMarkdown(filepath, { clockOffset } = { clockOffset: 0 }) {
+  if (!highlighter) {
```

Comment:
Make the shiki highlighter a long lived singleton — we shouldn't be creating it again per request!